domain/resource/state/resource.go

```diff
@@ -1781,26 +1782,42 @@ WHERE  resource_uuid = $update.old_uuid
 	rows, err := outcome.Result().RowsAffected()
 	if err != nil {
 		return errors.Capture(err)
+	} else if rows == 0 {
+		// There was a race of two users trying to change the same
+		// resource at the same time. Give a specific error message.
+		// On the client side, the end user could retry to the action
+		// or view the already updated resource.
+		return errors.Errorf("attempting to update old resource version.")
```

Comment:
This error should still include information about what actually went wrong, there is potential here that the callsite of the methods change in the future and this message becomes outdated and incorrect. It currently implicitly assumes something about the callers of this method.

It should still say what the symptom of the error is, i.e. that when updating the application resource we expected 1 changed and got 0.